### ISSUE TITLE
Changing the location of keepassxc-browser JSON files with portable KeePassXC

### ISSUE THREAD

**Issue by PaduaPlay** [keepassxreboot/keepassxc#10061]
## Current Behavior
KeePassXC v 2.7.6, Windows portable version still stores the browser json files in the \users directory -

C:\Users\***\AppData\Local\KeePassXC\org.keepassxc.keepassxc_browser_vivaldi.json

## Expected Behavior
Since it is a portable mode all files should be in the \KeePassXC program directory.

Is there a way to ensure that all files related to KeePassXC are stored only in the \KeePassXC directory?

Thanks.

**Comment by droidmonkey** [comment]
I don't think this can be changed because that json file must be statically located (ie it can't move around). That file is not for KeePassXC, its for the browsers themselves. Also note that the registry is modified to make this all work as well.

**Comment by PaduaPlay** [comment]
Thanks for the responses.

Since the idea of proper portability is that no footprint should be left on the host system, it would be better if these json files are kept in the KeePassXC program directory- they are not expected to be moved around, but kept in the same folder as the KeePassC executable and, if needed, the sub folder location can be defined in the .ini file.

Most browsers, recent, privacy conscious ones, come in a portable mode and they themselves do not store files in any place other than defined folders, anyway.

If at all required, the json can be stored in the extension directory with each browser that KeePassXC is linked to - for example, in the X:\..\Extensions\oboonakemofpalcgghocfoadofidjkkk\ folder.

Even this should be unnecessary, if stored in the program directory along with the a config ini entry pointing to it.

Just the KeePassXC executable + extension code needs to look for these jsons in the specified folder.

As for the registry entries - they should be rendered totally unnecessary - since they are usually unused and unnecessary for portable apps.

> .. json file must be statically located (ie it can't move around). That file is not for KeePassXC, its for the browsers themselves. Also note that the registry is modified to make this all work as well...

**Comment by droidmonkey** [comment]
Sorry but that is not how this works. On Windows, all browsers configure native messaging by looking for entries in the registry that point to a specific json file that tells them how to launch the external application. Both the registry entry and the json file must be in alignment for it to work.

Also, portable definitely does not mean "no footprint" as that is impossible to achieve in this scenario. Portable in our sense means "not installed" and you can run KeePassXC wherever it lives. That does not apply to the browser extension functionality though.

**Comment by PaduaPlay** [comment]
@droidmonkey, Appreciate your taking the time to respond in detail.

First, KeePassXC is an excellent piece of software and many thanks to the full team that works on it and for making it available.

Distinctions between 'no-install-needed' apps and 'portable' apps are, perhaps, philosophical and to each team their own..

Agree that it may not be possible to have no foot print in some cases, such as with the browser extensions.

Just tried to move the JSON to a directory inside the KeePassXC program directory and changed the registry entry HKEY_CURRENT_USER\SOFTWARE\Chromium\NativeMessagingHosts\org.keepassxc.keepassxc_browser to point to that directory. And it works fine.

Appreciate that you have closed this as completed, but, if I may suggest, if no foot print is impossible, minimal footprint option can be considered and, in this case, a setting somewhere can allow the user to store the JSONs in a specified directory - and make the registry entry accordingly. This can allow 'portability' of the KeePassXC app and its direct dependencies - of course, browser extension installation is outside the scope of this..

Thanks for considering.

**Comment by droidmonkey** [comment]
@varjolintu I'm ok with placing the json files in the portable config folder when in portable mode. However we should also enable the automatic update of native messaging configuration when in portable mode add well, similar to appimage.

**Comment by varjolintu** [comment]
@droidmonkey Sure. I'll do that.

**Comment by PaduaPlay** [comment]
Thank you, @droidmonkey and @varjolintu.
